eng/breakingChanges/breaking-change-doc.ps1

```diff
@@ -0,0 +1,985 @@
+    Write-Host "`n🧹 Cleaning previous data..." -ForegroundColor Yellow
+    if (Test-Path $outputRoot) { Remove-Item $outputRoot -Recurse -Force }
+    Write-Host "✅ Cleanup completed" -ForegroundColor Green
+}
+
+# Validate parameters
+if (-not $PrNumber -and -not $Query) {
+    Write-Error @"
+❌ Either -PrNumber or -Query must be specified.
+
+EXAMPLES:
+  Process specific PR:
+    .\breaking-change-doc.ps1 -PrNumber 114929
+
+  Query for PRs (example - customize as needed):
+    .\breaking-change-doc.ps1 -Query "repo:dotnet/runtime state:closed label:needs-breaking-change-doc-created is:merged merged:>2024-09-16 -milestone:11.0.0"
+
+Use -Help for more examples and detailed usage information.
+"@
+    exit 1
+}
+
+if ($PrNumber -and $Query) {
+    Write-Error "❌ Cannot specify both -PrNumber and -Query. Choose one."
+    exit 1
+}
+
+# Determine action mode - default to analysis only if no action specified
+$executeActions = $CreateIssues -or $Comment
+
+# Validate parameter combinations
+if (($CreateIssues -and $Comment) -or ($CollectOnly -and ($CreateIssues -or $Comment))) {
+    Write-Error "❌ Cannot combine -CollectOnly, -CreateIssues, and -Comment. Choose one action mode."
+    exit 1
+}
+
+$actionMode = if ($CollectOnly) { "Collect Only" }
+              elseif ($CreateIssues) { "Create Issues" }
+              elseif ($Comment) { "Add Comments" }
+              else { "Analysis Only" }
+
+Write-Host "   Action Mode: $actionMode" -ForegroundColor Cyan
+if (-not $executeActions -and -not $CollectOnly) {
+    Write-Host "   📝 Will generate drafts without making changes to GitHub" -ForegroundColor Yellow
+}
+
+# Function to safely truncate text
+function Limit-Text {
+    param([string]$text, [int]$maxLength = 2000)
+
+    if (-not $text -or $text.Length -le $maxLength) {
+        return $text
+    }
+
+    $truncated = $text.Substring(0, $maxLength)
+    $lastPeriod = $truncated.LastIndexOf('.')
+    $lastNewline = $truncated.LastIndexOf("`n")
+
+    $cutPoint = [Math]::Max($lastPeriod, $lastNewline)
+    if ($cutPoint -gt ($maxLength * 0.8)) {
+        $truncated = $truncated.Substring(0, $cutPoint + 1)
+    }
+
+    return $truncated + "`n`n[Content truncated for length]"
+}
+
+# Function to URL encode text for GitHub issue URLs
+function Get-UrlEncodedText {
+    param([string]$text)
+
+    # Basic URL encoding for GitHub issue URLs
+    $encoded = $text -replace '\r\n', '%0A' -replace '\n', '%0A' -replace '\r', '%0A'
+    $encoded = $encoded -replace ' ', '%20' -replace '#', '%23' -replace '&', '%26'
+    $encoded = $encoded -replace '\[', '%5B' -replace '\]', '%5D' -replace '\(', '%28' -replace '\)', '%29'
+    $encoded = $encoded -replace ':', '%3A' -replace ';', '%3B' -replace '\?', '%3F' -replace '=', '%3D'
+    $encoded = $encoded -replace '@', '%40' -replace '\+', '%2B' -replace '\$', '%24'
+    $encoded = $encoded -replace '"', '%22' -replace "'", '%27' -replace '<', '%3C' -replace '>', '%3E'
+
+    return $encoded
+}
```

Comment:
Replace by `[System.Web.HttpUtility]::UrlEncode($text)`?

Reply:
I think you might actually want [Uri.EscapeDataString](https://learn.microsoft.com/en-us/dotnet/api/system.uri.escapedatastring?view=net-9.0) here since you are only escaping a single element of data rather than the full Url itself.